src/data.rs

```diff
@@ -294,3 +549,41 @@ pub(crate) fn record_batch_from_data_and_columns(
 
     record_batch_from_name_array_iterator(iter, schema)
 }
+
+pub(crate) fn record_batch_from_data_and_columns_with_errors(
+    columns: &[ColumnInfo],
+    data: &ExcelSheetData,
+    offset: usize,
+    limit: usize,
+) -> FastExcelResult<(RecordBatch, CellErrors)> {
+    let schema = selected_columns_to_schema(columns);
+
+    let mut cell_errors = CellErrors { errors: vec![] };
+
+    let iter = columns.iter().map(|column_info| {
+        let col_idx = column_info.index();
+        let dtype = *column_info.dtype();
+
+        let (array, new_cell_errors) = match dtype {
+            DType::Null => (
+                Arc::new(NullArray::new(limit - offset)) as Arc<dyn arrow::array::Array>,
+                CellErrors { errors: vec![] },
+            ),
+            DType::Int => create_int_array_with_errors(data, col_idx, offset, limit),
+            DType::Float => create_float_array_with_errors(data, col_idx, offset, limit),
+            DType::String => create_string_array_with_errors(data, col_idx, offset, limit),
+            DType::Bool => create_boolean_array_with_errors(data, col_idx, offset, limit),
+            DType::DateTime => create_datetime_array_with_errors(data, col_idx, offset, limit),
+            DType::Date => create_date_array_with_errors(data, col_idx, offset, limit),
+            DType::Duration => create_duration_array_with_errors(data, col_idx, offset, limit),
+        };
+
+        cell_errors.errors.extend(new_cell_errors.errors);
```

Comment:
Oh actually you could just change the return type of the `create_array_with_errors` function to `(Arc<dyn Array>, Vec<CellError>)` Since you're only using the returned `CellErrors` struct to extend the struct here

Reply:
Updated